File: PHP/http/request/HttpRequest.php
<?php

abstract class HttpRequest {

    abstract function sendRequest($url);

    /**
     * Processes the array being passed and returns a properly formatted string for the HTTP request
     * @param array $data The data that will be sent as part of the request
     * @return string The data in the appropriate format for an HTTP request
     */
    protected function setData($data) {
        $requestString = "";
        foreach($data as $key => $value) {
            $requestString .= urlencode($key) . "=" . urlencode($value);
            $requestString .= "&";
        }
        $requestString = rtrim($requestString, "&");
        return $requestString;
    }

}

?>

File: Java/src/com/aaronjwood/multilib/http/request/HttpRequest.java
package com.aaronjwood.multilib.http.request;

import java.io.IOException;
import java.io.UnsupportedEncodingException;
import java.util.Map;

public abstract class HttpRequest {

    abstract String sendRequest(String url) throws IOException;

    abstract String setData(Map<String, String> data) throws UnsupportedEncodingException;

}
